mobile_app/test/test_trigger_build.py

```diff
@@ -0,0 +1,139 @@
+"""
+Tests for trigger app build script
+"""
+
+import git
+import json
+from mock import Mock, patch
+import os
+import tempfile
+from unittest import TestCase
+
+from mobile_app import trigger_build
+
+TEST_BRANCH = "test-branch"
+
+
+class Aborted(Exception):
+    """
+    Helper exception for indicating system exit instead of
+    actually exiting the process
+    """
+    pass
+
+
+def _trigger_build(repo_path, environ):
+    """
+    Helper to kick off the trigger build task
+    """
+    trigger_build.run_trigger_build(
+        [
+            "--repo-path", repo_path,
+            "--branch-name", TEST_BRANCH
+        ],
+        environ
+    )
+
+
+def _dummy_environ():
+    """
+    Generates a test config for all known keys
+    """
+    result = {}
+    for key in trigger_build.EXPECTED_ENVIRONMENT_VARIABLES:
+        result[key] = "test-value"
+    return result
+
+
+def _make_temp_repo():
+    """
+    Creates a temporary repository for testing
+    """
+
+    repo_path = tempfile.mkdtemp(prefix="test-repo")
```

Comment:
(addCleanup is preferable since that will be called even in the instance of a test error.)